packages/client/src/lib.js

```diff
@@ -216,13 +219,28 @@ class NFTStorage {
    * `foo/bla/baz.json` is ok but `foo/bar.png`, `bla/baz.json` is not.
    *
    * @param {Service} service
-   * @param {Iterable<File>|AsyncIterable<File>} files
+   * @param {FilesSource} filesSource
    * @returns {Promise<CIDString>}
    */
-  static async storeDirectory(service, files) {
+  static async storeDirectory(service, filesSource) {
     const blockstore = new Blockstore()
+    /** @type {AsyncIterable<File>} */
+    const files = pipe(
+      async function* () {
+        yield* filesSource
+      },
+      async function* (files) {
+        for await (const fileObject of files) {
+          /** @type {File} */
+          const file =
+            fileObject instanceof File
+              ? fileObject
+              : new File(await all(fileObject.stream()), fileObject.name)
```

Comment:
Do we need to `await` the stream here? It would be cool if we could just pass in `[fileObject.stream()]`

Reply:
I think that `toImportCandidate` inside of `encodeDirectory` will accept anything with a `stream()` method that returns an async iterable. So you may be able to skip the coercion to `File` here and just pass everything through.

Unless you tried that already and it didn't work, lol

Reply:
it wasn't so simple. I didn't want to augment more functions than needed, but insofar as somethings were typed as wanting `File`, this required buffering. afaict constructing a dom `File` requires not just a stream but a fully-read `Uint8Array`.

I just pushed a change though that allows the `FileObject` to `encodeDirectory` and makes this code less ugly.